corfudb-tools/src/main/java/org/corfudb/compactor/CorfuStoreCompactorMain.java

```diff
@@ -0,0 +1,336 @@
+            + "--keystore=<keystore_file> KeyStore File\n"
+            + "--ks_password=<keystore_password> KeyStore Password\n"
+            + "--truststore=<truststore_file> TrustStore File\n"
+            + "--truststore_password=<truststore_password> Truststore Password\n"
+            + "--persistedCacheRoot=<pathToTempDirForLargeTables> Path to Temp Dir\n"
+            + "--maxWriteSize=<maxWriteSize> Max write size smaller than 2GB\n"
+            + "--bulkReadSize=<bulkReadSize> Read size for chain replication\n"
+            + "--trim=<trim> Should trim be performed in this run\n"
+            + "--isUpgrade=<isUpgrade> Is this called during upgrade\n"
+            + "--trimTokenFile=<trimTokenFilePath> file to store the trim tokens during upgrade"
+            + "--upgradeDescriptorTable=<upgradeDescriptorTable> Repopulate descriptor table?\n"
+            + "--tlsEnabled=<tls_enabled>";
+
+    public CorfuStoreCompactorMain() {
+        CorfuRuntimeHelper corfuRuntimeHelper;
+        CorfuRuntimeHelper cpRuntimeHelper;
+        if (tlsEnabled) {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+            cpRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize,
+                    runtimeKeyStore, runtimeKeystorePasswordFile,
+                    runtimeTrustStore, runtimeTrustStorePasswordFile);
+
+        } else {
+            corfuRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+            cpRuntimeHelper = new CorfuRuntimeHelper(hostname, port, maxWriteSize, bulkReadSize);
+        }
+
+        corfuRuntime = corfuRuntimeHelper.getRuntime();
+        cpRuntime = cpRuntimeHelper.getRuntime();
+        corfuStore = new CorfuStore(corfuRuntime);
+        distributedCompactor = new DistributedCompactor(corfuRuntime, cpRuntime, persistedCacheRoot);
+
+        this.openCompactionTables();
+    }
+
+    private void upgrade() {
+        retryCheckpointing = 10;
+        if (upgradeDescriptorTable) {
+            syncProtobufDescriptorTable();
+        }
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            txn.putRecord(checkpointTable, DistributedCompactor.UPGRADE_KEY, TokenMsg.getDefaultInstance(),
+                    null);
+            txn.commit();
+        } catch (Exception e) {
+            log.warn("Unable to write UpgradeKey to checkpoint table, ", e);
+        }
+    }
+
+    /**
+     * Entry point to invoke Client checkpointing by the CorfuServer
+     *
+     * @param args command line argument strings
+     */
+    @SuppressWarnings("UncommentedMain")
+    public static void main(String[] args) throws Exception {
+        getCompactorArgs(args);
+
+        if (maxWriteSize == -1) {
+            if (persistedCacheRoot == null) {
```

Comment:
why it can be null? Why we can't initially create a variable with some predefined value?